cli_tools/common/utils/logging/logger.go

```diff
@@ -15,50 +15,179 @@
 package logging
 
 import (
-	"fmt"
 	"log"
+	"os"
+	"strings"
 	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/proto/go/pb"
+	"github.com/golang/protobuf/proto"
 )
 
-// LogEntry encapsulates a single log entry.
-type LogEntry struct {
-	LocalTimestamp time.Time `json:"localTimestamp"`
-	Message        string    `json:"message"`
+// LogWriter is a logger for interactive tools. It supports
+// string messages and structured metrics.
+//
+// Structured metrics are accumulated over the lifespan of the logger.
+//
+// To rebuild the mock, run `go generate ./...`
+//go:generate go run github.com/golang/mock/mockgen -package mocks -source $GOFILE -destination ../../../mocks/mock_logger.go
+type LogWriter interface {
+	// WriteUser messages are shown in Pantheon or Gcloud to the user.
```

Comment:
and also in OutputInfo.serial_outputs?
I feel there are overlaps among these 4 methods right? They are not all exclusive.

Reply:
serialized in the latest version.

Reply:
I meant the comment line - "WriteUser messages are shown in Pantheon or Gcloud to the user", per my understanding it's also passed into OutputInfo.serial_outputs, so I feel the description is not that accurate. Let me know if I misunderstood it.

Reply:
I see -- good point :)  Clarified in the latest version where each message will be shown.